interbtc-spec/docs/source/spec/nomination.rst

```diff
@@ -208,463 +131,154 @@ Specification
-
-``executeNominatorWithdrawal(nominatorId, operatorId)``
-
-*Parameters*
-
-* ``nominatorId``: the id of the requester.
-
-* ``operatorId``: the id of the operator.
-
-*Events*
-
-* ``ExecuteNominatorCollateralWithdrawal(nominatorId, operatorId, unbondedCollateral)``
-
-*Errors*
-
-* ``VaultNotOptedInToNomination``: the ``operatorId`` is not an Operator.
-
-* ``NoMaturedCollateral``: either no collateral withdrawal has been requested, or the requests have not matured yet.
-
-* ``NominatorNotFound``: the ``nominatorId`` is not a Nominator.
-
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
-
-1. Check ``operatorId`` is an operator. If not, throw ``VaultNotOptedInToNomination``.
-
-2. Check ``nominatorId`` is a nominator. If not, throw ``NominatorNotFound``.
-
-3. Iterate through the ``withdrawalRequests`` array in the ``Nominator`` struct inside the ``Operator`` struct for ``operatorId``. Determine how much collateral was unbonded, removing matured requests.
-
-4. If there is zero unbonded collateral, throw ``NoMaturedCollateral``.
+*Postconditions*
 
+* The vault's collateral MUST increase by the amount nominated.
+* The nominators MUST earn rewards equivalent to their collateral.
 
 
-.. _cancelOperatorWithdrawal:
+.. _withdrawNominationCollateral:
 
-cancelOperatorWithdrawal
-------------------------
-
-Cancel an operator's withdrawal request.
-
-Specification
-.............
-
-*Function Signature*
-
-``cancelOperatorWithdrawal(operatorId, requestId)``
-
-*Parameters*
-
-* ``operatorId``: the id of the operator.
-
-* ``requestId``: the id of the withdrawal request.
-
-*Events*
-
-* ``CancelOperatorCollateralWithdrawal(requestId, operatorId)``
-
-*Errors*
-
-* ``VaultNotOptedInToNomination``: the ``operatorId`` is not an Operator.
-
-* ``WithdrawalRequestNotFound``: no withdrawal request found for the given id.
-
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
-
-1. Check ``operatorId`` is an operator. If not, throw ``VaultNotOptedInToNomination``.
-
-2. Check ``requestId`` corresponds to an actual withdrawal request. If not, throw ``WithdrawalRequestNotFound``.
-
-3. Remove the withdrawal request from the ``withdrawalRequests`` array in the ``Operator`` struct for ``operatorId``.
-
-4. Increase the backing collateral of ``operatorId`` in the :ref:`Vault-registry` by the amount in the withdrawal request.
-
-
-.. _cancelNominatorWithdrawal:
-
-cancelNominatorWithdrawal
--------------------------
+withdrawCollateral
+------------------
 
-Cancel a nominator's withdrawal request.
+Withdraw collateral from a nominated Vault.
 
 Specification
 .............
 
 *Function Signature*
 
-``cancelNominatorWithdrawal(nominatorId, operatorId, requestId)``
+``withdrawCollateral(vaultId, nominatorId, amount)``
```

Comment:
(also below in the event) vaultId and NominatorId are swapped

Reply:
updated in the implementation